### PR TITLE
taup: Add tutorial for taup_time

### PR DESCRIPTION
**Pull request by seisman**
Related to #1.

Adapted from https://blog.seisman.info/taup-calculate-traveltime/

### CODE REVIEW

File: source/taup/taup_time.rst [hunk 1]
```diff
@@ -0,0 +1,159 @@
+taup_time
+=========
+
+TauP 提供的 ``taup_time`` 命令可用于计算震相走时、射线参数、出射角、
+入射角等信息。
+
+基本用法
+--------
+
+使用 ``taup_time -help`` 可查询该命令的完整用法和选项。此处仅介绍最常用用法。
```
Comment (core-man) [inline source/taup/taup_time.rst:10]:
```suggestion
使用 ``taup_time -help`` 可查询该命令的完整用法和选项，此处仅介绍常用用法。
```

File: source/taup/taup_time.rst [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+计算震源深度为 300 km，震中距 60° 处，P、S、PcP 等震相的走时信息::
+
+   $ taup_time -ph P,S,PcP,ScS,PKiKP -deg 60 -h 300
+
+-  ``-ph`` 指定了一系列以逗号分隔的震相名
+-  ``-h`` 指定震源深度（单位 km）
+-  ``-deg`` 指定震中距（单位为°）
```
Comment (core-man) [inline source/taup/taup_time.rst:18]:
```suggestion
计算震源深度为 300 km，震中距 60° 处，P、S、PcP、ScS、PKiKP 震相的走时信息::

   $ taup_time -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP

-  ``-h``：震源深度（单位 km）
-  ``-deg``：震中距（单位 °）
-  ``-ph``：以逗号分隔的震相名
```

File: source/taup/taup_time.rst [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+TauP 默认使用 iasp91 模型，可以使用 ``-mod`` 选项设置要使用的速度模型::
+
+   $ taup_time -ph P,S,PcP,ScS,PKiKP -deg 60 -h 300 -mod prem
```
Comment (core-man) [inline source/taup/taup_time.rst:22]:
```suggestion
TauP 默认使用 iasp91 模型，可以使用 ``-mod`` 选项设置其他 TauP 预定义速度模型::

   $ taup_time -mod prem -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP
```

File: source/taup/taup_time.rst [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+TauP 中默认台站位于地表处，也可使用 ``--stadepth`` 指定台站深度（单位为 km）::
+
+   $ taup_time -ph P,S,PcP,ScS,PKiKP -deg 60 -h 300 --stadepth 0.5
```
Comment (core-man) [inline source/taup/taup_time.rst:26]:
```suggestion
TauP 默认台站位于地表，可使用 ``--stadepth`` 选项设置台站深度（单位为 km）::

   $ taup_time -h 300 -deg 60 --stadepth 0.5 -ph P,S,PcP,ScS,PKiKP
```

File: source/taup/taup_time.rst [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+若已知震源和台站位置，也可以使用 ``-sta`` 和 ``-evt`` 分别指定台站和震源的
+纬度和经度。例如，台站位于 (20°N,130°E)，震源位于 (40°N,150°E)，则::
+
+   $ taup_time -mod prem -ph P,S,PcP,ScS,PKiKP -h 100 -sta 20 130 -evt 40 170
```
Comment (core-man) [inline source/taup/taup_time.rst:31]: （20, 130）& （80,150）has a distance of about 60 degrees.

```suggestion
若已知震源和台站位置，也可以使用 ``-evt`` 和 ``-sta`` 选项分别指定震源和台站的
纬度和经度。例如，震源位于 (80 °N，150 °E)，台站位于 (20 °N，130 °E)::

   $ taup_time -mod prem -h 300 -evt 80 150 -sta 20 130 -ph P,S,PcP,ScS,PKiKP
```

File: source/taup/taup_time.rst [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+在震中距比较小的情况下，可以使用 ``-km`` 选项指定震中距（单位为 km）。
+比如计算一个浅源地震在 100 km 处常见 P 波震相的走时信息::
+
+   $ taup_time -ph P,S,PcP,ScS,PKiKP -h 5 -km 100
```
Comment (core-man) [inline source/taup/taup_time.rst:36]:
```suggestion
在震中距比较小的情况下，可以使用 ``-km`` 选项指定震中距（单位为 km）。
比如计算震源深度为 5 km，震中距 100 km 处，P,S,PcP,ScS,PKiKP 震相的走时信息::

   $ taup_time -h 5 -km 100 -ph P,S,PcP,ScS,PKiKP
```

File: source/taup/taup_time.rst [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+5.  射线参数（秒/度）
+
+    .. note::
+
+        射线参数有三种常用单位：秒/度（s/deg ）、秒/弧度（s/radian）、秒/千米（s/km）。
+        三者之间的关系为（:math:`R_0` 为地球半径）：
+
+        .. math::
+
+            1 s/deg = \frac{1}{\pi*R_0/180} s/km = 1/111.2 s/km
```
Comment (core-man) [inline source/taup/taup_time.rst:68]:
```suggestion
        三者之间的关系为（:math:`R_0` 为地球半径，假设为 6371 km）：

        .. math::

            1 s/deg = \frac{1}{\pi*R_0/180} s/km = 1/111.1949 s/km
```

File: source/taup/taup_time.rst [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+
+        .. math::
+
+            1 s/deg = \frac{1}{\pi / 180} s/radian = 57.3 s/radian
```
Comment (core-man) [inline source/taup/taup_time.rst:72]:
```suggestion
            1 s/deg = \frac{1}{\pi/180} s/radian = 57.29578 s/radian
```

File: source/taup/taup_time.rst [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+
+6.  出射角（度）：即射线从震源出射时与\ **垂直向下**\ 方向的夹角，取值范围为0到180°。
+
+    - 出射角 0° 到 90° 内表示射线向下出射（例如P震相）
+    - 出射角 90° 到 180° 内表示射线向上出射（例如pP震相)
```
Comment (core-man) [inline source/taup/taup_time.rst:77]:
```suggestion
6.  出射角（度）：即射线从震源出射时与\ **垂直向下**\ 方向的夹角，取值范围为 0 到 180°。

    - 0° 到 90° 表示射线向下出射（例如 P 震相）
    - 90° 到 180° 表示射线向上出射（例如 pP 震相)
```

File: source/taup/taup_time.rst [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+
+    .. note::
+
+       不同文章对出射角的定义不同，使用时应格外注意。
+
+       老版本的TauP对出射角的定义不同，可视为BUG，应避免使用老版本TauP。
```
Comment (core-man) [inline source/taup/taup_time.rst:83]:
```suggestion
       老版本的 TauP 对出射角的定义不同，可视为 BUG，应避免使用老版本 TauP。
```

File: source/taup/taup_time.rst [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+
+7.  入射角（度）：射线在台站处入射时与\ **垂直向上**\ 方向的夹角。
```
Comment (core-man) [inline source/taup/taup_time.rst:85]:
```suggestion
7.  入射角（度）：射线入射到台站时与\ **垂直向上**\ 方向的夹角。
```

File: source/taup/taup_time.rst [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+基本用法
+--------
+
+使用 ``taup_time -help`` 可查询该命令的完整用法和选项，此处仅介绍常用用法。
+
+计算震源深度为 300 km，震中距 60° 处，P、S、PcP、ScS、PKiKP 震相的走时信息::
+
+   $ taup_time -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP
+
+-  ``-h``：震源深度（单位 km）
```
Comment (core-man) [inline source/taup/taup_time.rst:16]:
```suggestion
-  ``-h``：震源深度（单位 km），默认深度为 0.0 km。
```

File: source/taup/taup_time.rst [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+
+       老版本的TauP对出射角的定义不同，可视为BUG，应避免使用老版本TauP。
+
+7.  入射角（度）：射线在台站处入射时与\ **垂直向上**\ 方向的夹角。
+8.  Purist 距离：地震波实际走过的圆弧距离。
+
+    该值可能与震中距不同，例如 50° 震中距处记录到的 PKKKP 震相，实际走过的
+    圆弧距离为 410°::
+
+        $ taup_time -ph PKKKP -deg 50
```
Comment (core-man) [inline source/taup/taup_time.rst:91]:
```suggestion
        $ taup_time -deg 50 -ph PKKKP
```

File: source/taup/taup_time.rst [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+        -----------------------------------------------------------------------------------
+          50.00     0.0   PKKKP   2519.56     2.242      6.72     6.72   410.00   = PKKKP
+
+9.  Purist 震相名：地震波的“真实”震相名。
+
+    例如，若想要计算 SS 震相在 660 km 不连续面的下表面的反射波走时，可指定其震相名为
+    ``S^660S``。由于 PREM 模型中 660 km 不连续面的“真实”深度为670 km，此时
+    TauP 实际返回的时从 670 km 处反射回的震相，此时 Purist 震相名会显示为 ``S^670S``::
+
+        $ taup_time -mod prem -ph S^660S -deg 120
```
Comment (core-man) [inline source/taup/taup_time.rst:105]:
```suggestion
    例如，若想要计算 SS 震相在 660 km 不连续面的下表面的反射波走时，可指定其震相名为
    ``S^660S``。由于 PREM 模型中 660 km 不连续面的“真实”深度为670 km，此时
    TauP 实际返回的时从 670 km 处反射回的震相，此时 Purist 震相名会显示为 ``S^670S``::

        $ taup_time -mod prem -deg 120 -ph S^660S
```

File: source/taup/taup_time.rst [hunk 15]
```diff
@@ -0,0 +1,159 @@
...
+        Model: prem
+        Distance   Depth   Phase    Travel    Ray Param  Takeoff  Incident  Purist    Purist
+          (deg)     (km)   Name     Time (s)  p (s/deg)   (deg)    (deg)   Distance   Name
+        ------------------------------------------------------------------------------------
+          120.00     0.0   S^660S   1975.75    12.336     20.79    20.79   120.00   * S^670S
+
+当使用 ``--rel`` 选项时，``taup_time`` 可输出相对于某个指定震相的走时差。例如，
+下面的命令中，最后一列输出了所有震相相对于 PcP 震相的走时差::
+
+    $ taup_time -ph P,S,PcP,ScS,PKiKP -deg 60 -h 300 -mod prem --rel PcP
```
Comment (core-man) [inline source/taup/taup_time.rst:116]:
```suggestion
可以使用 ``--rel`` 选项输出相对于某个指定震相的走时差。例如，
下面的命令中，最后一列输出了所有震相相对于 PcP 震相的走时差::

    $ taup_time -mod prem -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP --rel PcP
```

File: source/taup/taup_time.rst [hunk 16]
```diff
@@ -0,0 +1,159 @@
...
+       60.00   300.0   PcP      617.05     4.016     19.28    12.09    60.00   = PcP      0.00
+       60.00   300.0   PKiKP    993.45     1.245      5.88     3.72    60.00   = PKiKP  376.41
+       60.00   300.0   S       1043.98    12.659     34.16    21.37    60.00   = S      426.93
+       60.00   300.0   ScS     1134.89     7.480     19.38    12.43    60.00   = ScS    517.85
+
+信息提取
+--------
+
+上面介绍的输出中包含了很多信息，想要在脚本中提取出想要的信息有些麻烦。所以
+``taup_time`` 提供了一些选项，使得在脚本中计算走时什么的更加方便。
```
Comment (core-man) [inline source/taup/taup_time.rst:132]:
```suggestion
上面介绍的输出中包含了很多信息，想要在脚本中提取出想要的信息有些麻烦。所以
``taup_time`` 提供了一些选项，使得在脚本中计算走时、射线参数等更加方便。
```

File: source/taup/taup_time.rst [hunk 17]
```diff
@@ -0,0 +1,159 @@
...
+
+走时
+~~~~
```
Comment (core-man) [inline source/taup/taup_time.rst:135]: I usually use ++++ to indicate the third level, so let's make it a rule to use ~~~ now?

Comment (seisman) [reply source/taup/taup_time.rst:135]: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections

Comment (seisman) [reply source/taup/taup_time.rst:135]: Should we follow this?

File: source/taup/taup_time.rst [hunk 18]
```diff
@@ -0,0 +1,159 @@
...
+
+``--time`` 选项使得输出中只包含走时信息::
+
+   $ taup_time -mod prem -ph P,S,PcP -deg 60 -h 100 --time
+   595.3896 640.09875 1081.2472
```
Comment (core-man) [inline source/taup/taup_time.rst:140]:
```suggestion
``--time`` 选项使得输出中只包含走时信息，并且小数点的位数更多::

   $ taup_time -mod prem -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP --time
   574.7245 617.047 993.4524 1043.9805 1134.8923
```

File: source/taup/taup_time.rst [hunk 19]
```diff
@@ -0,0 +1,159 @@
...
+
+需要注意的是，输出中走时不是按照 ``-ph`` 选项中震相的顺序进行排序的，
+而是按照走时递增的顺序排序。比如这里，S 震相放在震相列表的第二位，
+而 S 震相的走时 1081.2472 却放在第三位。
```
Comment (core-man) [inline source/taup/taup_time.rst:144]:
```suggestion
需要注意的是，输出中走时不是按照 ``-ph`` 选项中震相的顺序进行排序的，
而是按照走时递增的顺序排序。比如这里，S 震相放在震相列表的第二位，
而 S 震相的走时 1043.9805 却放在第四位。
```

File: source/taup/taup_time.rst (hunk 17 above)
Comment (core-man) [reply source/taup/taup_time.rst:135]: # with overline, for parts
* with overline, for chapters

What do these two mean?

We may only need the following three terms:
=, for sections
-, for subsections
^, for subsubsections

Comment (seisman) [reply source/taup/taup_time.rst:135]: > What do these two mean?
> 
> * "# with overline", for parts
> * "* with overline", for chapters

They're just higher level headings.



> We may only need the following three terms:
> =, for sections
> -, for subsections
> ^, for subsubsections

I think yes.

File: source/taup/taup_time.rst [hunk 20]
```diff
@@ -0,0 +1,159 @@
...
+
+射线参数
+~~~~~~~~
+
+``--rayp`` 选项使得输出中只包含射线参数信息::
+
+   $ taup_time -mod prem -ph P,S,PcP -deg 60 -h 100 --rayp
+   6.8185554 3.9990747 12.785722
+
+此时，射线参数的输出是按照震相进行排序的。
```
Comment (core-man) [inline source/taup/taup_time.rst:154]:
```suggestion
``--rayp`` 选项使得输出中只包含射线参数信息，并且小数点的位数更多::

   $ taup_time taup_time -mod prem -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP --rayp
   6.739042 4.0158706 1.2450149 12.65922 7.480467

此时，输出的射线参数是按照 ``-ph`` 选项中震相的顺序进行排序的。
```

File: source/taup/taup_time.rst [hunk 21]
```diff
@@ -0,0 +1,159 @@
...
+
+走时
+~~~~
+
+``--time`` 选项使得输出中只包含走时信息，并且小数点的位数更多::
+
+   $ taup_time -mod prem -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP --time
+   574.7245 617.047 993.4524 1043.9805 1134.8923
+
+需要注意的是，输出中走时不是按照 ``-ph`` 选项中震相的顺序进行排序的，
```
Comment (core-man) [inline source/taup/taup_time.rst:142]:
```suggestion
需要注意的是，输出的走时不是按照 ``-ph`` 选项中震相的顺序进行排序的，
```

File: source/taup/taup_time.rst [hunk 22]
```diff
@@ -0,0 +1,159 @@
...
+       60.00   300.0   ScS     1134.89     7.480     19.38    12.43    60.00   = ScS    517.85
+
+信息提取
+--------
+
+上面介绍的输出中包含了很多信息，想要在脚本中提取出想要的信息有些麻烦。所以
+``taup_time`` 提供了一些选项，使得在脚本中计算走时、射线参数等更加方便。
+
+走时
+~~~~
```
Comment (core-man) [inline source/taup/taup_time.rst:135]:
```suggestion
走时
^^^^^
```

File: source/taup/taup_time.rst [hunk 23]
```diff
@@ -0,0 +1,159 @@
...
+
+   $ taup_time -mod prem -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP --time
+   574.7245 617.047 993.4524 1043.9805 1134.8923
+
+需要注意的是，输出中走时不是按照 ``-ph`` 选项中震相的顺序进行排序的，
+而是按照走时递增的顺序排序。比如这里，S 震相放在震相列表的第二位，
+而 S 震相的走时 1043.9805 却放在第四位。
+
+射线参数
+~~~~~~~~
```
Comment (core-man) [inline source/taup/taup_time.rst:147]:
```suggestion
射线参数
^^^^^^^^^
```

File: source/taup/taup_time.rst [hunk 24]
```diff
@@ -0,0 +1,159 @@
...
+
+   $ taup_time -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP
+
+-  ``-h``：震源深度（单位 km），默认深度为 0.0 km。
+-  ``-deg``：震中距（单位 °）
+-  ``-ph``：以逗号分隔的震相名
+
+TauP 默认使用 iasp91 模型，可以使用 ``-mod`` 选项设置其他 TauP 预定义速度模型::
+
+   $ taup_time -mod prem -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP
```
Comment (core-man) [inline source/taup/taup_time.rst:22]:
```suggestion
   $ taup_time -mod prem -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP
   $ taup_time -mod ak135 -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP
```

File: source/taup/taup_time.rst [hunk 25]
```diff
@@ -0,0 +1,160 @@
...
+   $ taup_time -mod ak135 -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP
+
+TauP 默认台站位于地表，可使用 ``--stadepth`` 选项设置台站深度（单位为 km）::
+
+   $ taup_time -h 300 -deg 60 --stadepth 0.5 -ph P,S,PcP,ScS,PKiKP
+
+若已知震源和台站位置，也可以使用 ``-evt`` 和 ``-sta`` 选项分别指定震源和台站的
+纬度和经度。例如，震源位于 (80 °N，150 °E)，台站位于 (20 °N，130 °E)::
+
+   $ taup_time -mod prem -h 300 -evt 80 150 -sta 20 130 -ph P,S,PcP,ScS,PKiKP
```
Comment (core-man) [inline source/taup/taup_time.rst:32]:
```suggestion
若已知震源和台站位置，也可以使用 ``-evt`` 和 ``-sta`` 选项分别指定震源和台站的
纬度和经度。例如，震源位于 (40 °N，150 °E)，台站位于 (20 °S，130 °E)::

   $ taup_time -mod prem -h 300 -evt 40 150 -sta -20 130 -ph P,S,PcP,ScS,PKiKP
```

File: source/taup/taup_time.rst [hunk 26]
```diff
@@ -0,0 +1,160 @@
...
+   $ taup_time -mod ak135 -h 300 -deg 60 -ph P,S,PcP,ScS,PKiKP
+
+TauP 默认台站位于地表，可使用 ``--stadepth`` 选项设置台站深度（单位为 km）::
+
+   $ taup_time -h 300 -deg 60 --stadepth 0.5 -ph P,S,PcP,ScS,PKiKP
+
+若已知震源和台站位置，也可以使用 ``-evt`` 和 ``-sta`` 选项分别指定震源和台站的
+纬度和经度。例如，震源位于 (40 °N，10 °E)，台站位于 (20 °S，10 °W)::
+
+   $ taup_time -mod prem -h 300 -evt 40 10 -sta -20 -10 -ph P,S,PcP,ScS,PKiKP
```
Comment (core-man) [inline source/taup/taup_time.rst:32]:
```suggestion
   $ taup_time -h 300 -evt 40 10 -sta -20 -10 -ph P,S,PcP,ScS,PKiKP
```

File: source/taup/taup_time.rst [hunk 27]
```diff
@@ -0,0 +1,160 @@
...
+
+   $ taup_time -h 5 -km 100 -ph P,S,PcP,ScS,PKiKP
+
+输出格式
+--------
+
+下面的命令计算了 PREM 模型下，震源深度 100 km，震中距 60° 处，P 和 pP 震相的走时信息。
+其输出为::
+
+   $ taup_time -mod prem -h 100 -deg 60 -ph pP,P
```
Comment (core-man) [inline source/taup/taup_time.rst:45]:
```suggestion
   $ taup_time -mod prem -h 100 -deg 60 -ph P,pP
```

File: source/taup/taup_time.rst [hunk 28]
```diff
@@ -0,0 +1,160 @@
...
+       60.00   300.0   PcP      617.05     4.016     19.28    12.09    60.00   = PcP      0.00
+       60.00   300.0   PKiKP    993.45     1.245      5.88     3.72    60.00   = PKiKP  376.41
+       60.00   300.0   S       1043.98    12.659     34.16    21.37    60.00   = S      426.93
+       60.00   300.0   ScS     1134.89     7.480     19.38    12.43    60.00   = ScS    517.85
+
+信息提取
+--------
+
+上面介绍的输出中包含了很多信息，想要在脚本中提取出想要的信息有些麻烦。所以
+``taup_time`` 提供了一些选项，使得在脚本中计算走时、射线参数等更加方便。
```
Comment (core-man) [inline source/taup/taup_time.rst:133]:
```suggestion
``taup_time`` 提供了一些选项，使得在脚本中计算走时、射线参数更加方便。
```

**Review by core-man — APPROVED** [review]

**Pull request merged**